tools/schemacode/bidsschematools/render.py

```diff
@@ -257,36 +327,86 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
+                for suffix in group["suffixes"]:
+                    # The glossary indexes by the suffix identifier (TwoPE instead of 2PE),
+                    # but the rules reference the actual suffix string (2PE instead of TwoPE),
+                    # so we need to look it up.
+                    suffix_id = [
+                        k for k, v in schema["objects"]["suffixes"].items() if v["value"] == suffix
+                    ][0]
+
+                    suffix_string = utils._link_with_html(
+                        suffix,
+                        html_path=glossary_path,
+                        heading=f"{suffix_id}-suffixes",
+                        pdf_format=pdf_format,
+                    )
+                    strings.append(f"{string}_{suffix_string}")
 
             # Add extensions
             full_strings = []
             extensions = group["extensions"]
             extensions = [ext if ext != "*" else ".<extension>" for ext in extensions]
-            extensions = utils.combine_extensions(extensions)
             if len(extensions) >= n_dupes_to_combine:
                 # Combine exts when there are many, but keep JSON separate
                 if ".json" in extensions:
                     extensions = [".<extension>", ".json"]
                 else:
                     extensions = [".<extension>"]
 
+            ext_headings = []
+            for extension in extensions:
+                # The glossary indexes by the extension identifier (niigz instead of .nii.gz),
+                # but the rules reference the actual suffix string (.nii.gz instead of niigz),
+                # so we need to look it up.
+                ext_id = [
+                    k
+                    for k, v in schema["objects"]["extensions"].items()
+                    if v["value"] == extension
+                ]
+                if ext_id:
+                    ext_id = ext_id[0]
+                    ext_headings.append(f"{ext_id}-extensions")
+                else:
+                    ext_headings.append("extension-common_principles")
+
+            extensions = utils.combine_extensions(
+                extensions,
+                html_path=glossary_path,
+                heading_lst=ext_headings,
+                pdf_format=pdf_format,
+            )
+
             for extension in extensions:
                 for string in strings:
-                    new_string = string + extension
+                    new_string = f"{string}{extension}"
                     full_strings.append(new_string)
 
             full_strings = sorted(full_strings)
             if full_strings:
                 paragraph += "\n".join(full_strings) + "\n"
 
     paragraph = paragraph.rstrip()
-    codeblock = "Template:\n```Text\n" + paragraph + "\n```"
+    if pdf_format:
+        codeblock = f"Template:\n```Text\n{paragraph}\n```"
+    else:
+        codeblock = f'Template:\n<pre style="overflow-x:scroll;">\n{paragraph}\n</pre>'
```

Comment:
We lost the gray background.

```suggestion
        codeblock = f'Template:\n<pre style="overflow-x:scroll;"><code>\n{paragraph}\n</code></pre>'
```

Reply:
Good catch! I didn't even notice that.